src/mapgen.cpp

```diff
@@ -1981,13 +1979,13 @@ void mapgen_palette::load_place_mapings( const JsonObject &jo, const std::string
     if( member_name == "terrain" || member_name == "furniture" ) {
         return;
     }
-    JsonObject pjo = jo.get_object( member_name );
-    for( auto &key : pjo.get_member_names() ) {
+    for( const JsonMember member : jo.get_object( member_name ) ) {
+        const std::string key = member.name();
```

Comment:
Clang complains about this kind of assignment (unnecessary copy instead of a const ref) here and in two other places. I guess you don't need a temporary at all. Inlined `member.name()` wherever they occur would be cleaner.